src/rtracer.js

```diff
@@ -14,33 +14,49 @@ const wrapHttpEmitters = (req, res) => {
   wrapEmitter(res, asyncResource)
 }
 
+const expressSetResHeaderFnName = 'set'
+const fastifySetResHeaderFnName = 'setHeader'
+
 /**
- * Generates a request tracer middleware for Express.
- *
- * @param {Object} options possible options
- * @param {boolean} options.useHeader respect request header flag
- *                                    (default: `false`)
- * @param {string} options.headerName request header name, used if `useHeader` is set to `true`
- *                                    (default: `X-Request-Id`)
- * @param {function} options.requestIdFactory function used to generate request ids
- *                                    (default: UUIDs v1)
+ * Generates a function to generate tracer middleware for Express/Fastify.
+ * @param setResHeaderFnName {string} name of a function to set response header
  */
-const expressMiddleware = ({
-  useHeader = false,
-  headerName = 'X-Request-Id',
-  requestIdFactory = uuidv1
-} = {}) => {
-  return (req, res, next) => {
-    let requestId
-    if (useHeader) {
-      requestId = req.headers[headerName.toLowerCase()]
-    }
-    requestId = requestId || requestIdFactory()
+const expressMiddleware = (setResHeaderFnName) => {
```

Comment:
nit: a function could be used instead of the function name. Say, express middleware generation would look like the following:
```js
expressMiddleware((res, headerName, requestId) => res.set(headerName, requestId))
```